src/app/cli/src/init/client.ml

```diff
@@ -1037,36 +1069,40 @@ let pending_snark_work =
        yet"
     (Cli_lib.Background_daemon.graphql_init (return ())
        ~f:(fun graphql_endpoint () ->
-         Deferred.map
-           (Graphql_client.query_exn
-              Graphql_queries.Pending_snark_work.(make @@ makeVariables ())
-              graphql_endpoint )
-           ~f:(fun response ->
-             let lst =
-               [%to_yojson: Cli_lib.Graphql_types.Pending_snark_work.t]
-                 (Array.map
-                    ~f:(fun bundle ->
-                      Array.map bundle.workBundle ~f:(fun w ->
-                          let fee_excess_left = w.fee_excess.feeExcessLeft in
-                          { Cli_lib.Graphql_types.Pending_snark_work.Work
-                            .work_id = w.work_id
-                          ; fee_excess =
-                              Currency.Amount.Signed.of_fee
-                                (to_signed_fee_exn fee_excess_left.sign
-                                   fee_excess_left.feeMagnitude )
-                          ; supply_increase = w.supply_increase
-                          ; source_first_pass_ledger_hash =
-                              w.source_first_pass_ledger_hash
-                          ; target_first_pass_ledger_hash =
-                              w.target_first_pass_ledger_hash
-                          ; source_second_pass_ledger_hash =
-                              w.source_second_pass_ledger_hash
-                          ; target_second_pass_ledger_hash =
-                              w.target_second_pass_ledger_hash
-                          } ) )
-                    response.pendingSnarkWork )
-             in
-             print_string (Yojson.Safe.to_string lst) ) ) )
+         format_pending_snark_work graphql_endpoint ) )
+
+let pending_snark_work_range =
+  let open Command.Param in
+  let start_idx =
+    flag "--start-idx" ~aliases:[ "start-idx" ]
+      ~doc:"START_IDX first index of the range (included)"
+      (required Cli_lib.Arg_type.int16)
+  in
+  let end_idx =
+    flag "--end-idx" ~aliases:[ "end-idx" ]
+      ~doc:"END_IDX first index of the range (excluded)"
+      (optional Cli_lib.Arg_type.int16)
```

Comment:
*nit*
i still prefer to enforce this as positive through the 
type system instead of runtime, but your logic is good so I will approve. I’m also generally more a fan of returning empty list when the user inputs an extremely high end index instead of treating as a none case, but again the logic is fine.